btm/src/test/java/bitronix/tm/timer/TaskSchedulerTest.java

```diff
@@ -73,6 +73,19 @@ public void testTaskOrdering() throws Exception {
         assertEquals(2, result.get(2).getObject());
     }
 
+    public void testIdenticalScheduleTimestamp() throws Exception {
+        List<SimpleTask> result = Collections.synchronizedList(new ArrayList<SimpleTask>());
+
+        ts.addTask(new SimpleTask(new Date(MonotonicClock.currentTimeMillis() + 100), ts, 0, result));
+        ts.addTask(new SimpleTask(new Date(MonotonicClock.currentTimeMillis() + 200), ts, 1, result));
+        ts.addTask(new SimpleTask(new Date(MonotonicClock.currentTimeMillis() + 200), ts, 2, result));
```

Comment:
There is no real guarantee that there will be two identical timestamps since you're calling `currentTimeMillis()` three times.

Storing the 'common' timestamp in a local variable then passing it to the two tasks would make this test more robust.